From e70f330af74e902d4f1b1f4891a34d315cc09963 Mon Sep 17 00:00:00 2001
From: Gary Mirams <gary.mirams@gmail.com>
Date: Mon, 11 Aug 2014 14:41:49 +0100
Subject: [PATCH] Update the #includes in shared_ptr_helper.hpp to avoid
 'singleton' is not a member of 'boost::serialization' errors on compilation

---
 include/boost/serialization/shared_ptr_helper.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/boost/serialization/shared_ptr_helper.hpp b/boost/serialization/shared_ptr_helper.hpp
index 9dace87..78995d4 100644
--- a/boost/serialization/shared_ptr_helper.hpp
+++ b/boost/serialization/shared_ptr_helper.hpp
@@ -26,6 +26,8 @@
 #include <boost/type_traits/is_polymorphic.hpp>
 #include <boost/mpl/if.hpp>
 
+#include <boost/serialization/singleton.hpp>
+#include <boost/serialization/extended_type_info.hpp>
 #include <boost/serialization/type_info_implementation.hpp>
 #include <boost/serialization/throw_exception.hpp>
 #include <boost/archive/archive_exception.hpp>
